data/1-dlcs/dlc.js

```diff
@@ -0,0 +1,108 @@
+
+let dlcId = 1;
+const allUserTags = userTagsStr.split(',');
+const reviewRatings = ['Overwhelmingly Positive', 'Very Positive', 'Positive', 'Mostly Positive', 'Mixed', 'Mostly Negative', 'Negative', 'Very Negative', 'Overwhelmingly Negative'];
+let dlcs = [];
+const discounts = [0, 0, 0, 0, 0.10, 0.15, 0.20, 0.25, 0.5, 0.67, 0.75, 0.90];
+const maxDlc = 10;
+const maxDlcPrice = 50;
+const minDlcPrice = 1;
+const maxTags = 15;
+const minTags = 1;
+const maxPreviews = 4;
+const minPreviews = 1;
+const maxReviews = 100000;
+const minYear = 98;
+//  year of first game sold on steam
+
+
+
+
+
+/**
+ * In place shuffle for arrays
+ *
+ * @param {Array} arr Array to be shuffled
+ * @returns {Array} Array that has been shuffled
+ */
+const randomizeArr = function (arr) {
+  //  fisher-yates shuffle
+  let n = arr.length;
+  while (n) {
+    n -= 1;
+    const idx = Math.floor(Math.random() * n);
+    const temp = arr[idx];
+    arr[idx] = arr[n];
+    arr[n] = temp;
+  }
+  return arr;
+};
+
+/**
+ * Generates n number of dlcs for a particular game id
+ *
+ * @param {Int} associatedGameId
+ * @param {Int} n number of dlcs
+ * @returns {Array} Array with n dlcs
+ */
+const generateDlc = function (associatedGameId, n) {
+  let generatedDlcs = [];
+  for (let i = 0; i < n; i++) {
+    const dlc = {};
+    let numTags = Math.floor(Math.random() * (maxTags - minTags)) + minTags;
+    let numPreviews = Math.floor(Math.random() * (maxPreviews - minPreviews)) + minPreviews;
+
+    dlc.associatedGameId = associatedGameId;
+    dlc.dlcId = dlcId;
+    dlc.title = faker.fake('{{commerce.productName}}');
+    dlc.dlcPrice = Math.floor(Math.random() * (maxDlcPrice - minDlcPrice)) + minDlcPrice;
+    dlc.discountRate = discounts[Math.floor(Math.random() * discounts.length)];
+    dlc.releaseDate = faker.date.between(new Date(minYear, 0), Date()).toString();
+    dlc.totalReviews = Math.floor(Math.random() * maxReviews);
+    dlc.review_summary = reviewRatings[Math.floor(Math.random() * reviewRatings.length)];
+    dlc.userTags = [];
+    dlc.previews = [];
+
+    while (numTags) {
+      const idx = Math.floor(Math.random() * allUserTags.length);
+      const tag = allUserTags[idx];
+      if (!dlc.userTags.includes(tag)) {
+        dlc.userTags.push(tag);
+        numTags -= 1;
+      }
+    }
+    while (numPreviews) {
+      //get 600x338 image of random food from lorempixel
+      const url = faker.image.imageUrl(600, 338, 'food', true);
+      dlc.previews.push(url);
+      numPreviews -= 1;
+    }
+
+
+
+    dlcId += 1;
+    generatedDlcs.push(dlc);
+  }
+  return generatedDlcs;
+};
+
+
+
+const init = function () {
+  randomizeArr(discounts);
+  randomizeArr(reviewRatings);
+  randomizeArr(allUserTags);
```

Comment:
This also applies to the lines in your `generateDlc` function: If you're shuffling these data arrays here, why are you also getting a random index from these arrays inside `generateDlc`? Seems like a bit of unnecessary overhead.